ledger/eval.go

```diff
@@ -382,16 +383,21 @@ type ledgerForCowBase interface {
 // of the block that the caller is planning to evaluate. If the length of the
 // payset being evaluated is known in advance, a paysetHint >= 0 can be
 // passed, avoiding unnecessary payset slice growth.
```

Comment:
I was thinking about that. The reason I ended up with the current approach was to avoid creating "branching". I think that this new field isn't any different from the "hint", so it might be there as well.
I also considered creating the evaluator, and then calling a method on the evaluator to update this variable. Again, I think that it would be simpler from readability perspective to have this parameter on the `StartEvaluator`.

If we will end up passing too many optional arguments to the evaluator's constructor, I would create a struct for these.. but it doesn't look like we're there yet.